src/utils/openmc/material.py

```diff
@@ -0,0 +1,491 @@
+#!/usr/bin/env python
+
+import warnings
+
+import openmc
+from openmc.checkvalue import *
+from openmc.clean_xml import *
+from xml.etree import ElementTree as ET
+from collections import MappingView
+from copy import deepcopy
+import numpy as np
+
+
+# A list of all IDs for all Materials created
+MATERIAL_IDS = list()
+
+# A static variable for auto-generated Material IDs
+AUTO_MATERIAL_ID = 10000
+
+def reset_auto_material_id():
+    global AUTO_MATERIAL_ID, MATERIAL_IDS
+    AUTO_MATERIAL_ID = 10000
+    MATERIAL_IDS = list()
+
+
+# Units for density supported by OpenMC
+DENSITY_UNITS = ['g/cm3', 'g/cc', 'kg/cm3', 'at/b-cm', 'at/cm3', 'sum']
+
+# ENDF temperatures
+ENDF_TEMPS = np.array([300, 600, 700, 900, 1200, 1500])
+
+# ENDF ZAIDs
+ENDF_ZAIDS = np.array(['70c', '71c', '72c', '73c', '74c'])
+
+# Constant for density when not needed
+NO_DENSITY = 99999.
+
+
+class Material(object):
+
+    def __init__(self, material_id=None, name=''):
+
+        # Initialize class attributes
+        self._id = None
+        self._name = ''
+        self._density = None
+        self._density_units = ''
+
+        # A dictionary of Nuclides
+        # Keys         - Nuclide names
+        # Values     - tuple (nuclide, percent, percent type)
+        self._nuclides = dict()
```

Comment:
Fixed in PR by @paulromano. I personally like the less idiomatic but more transparent constructors for collections, but it's a minor and moot point.